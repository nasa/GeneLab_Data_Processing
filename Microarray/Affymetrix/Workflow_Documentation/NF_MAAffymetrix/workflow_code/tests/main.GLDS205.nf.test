nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("GLDS-205: Should run without failures") {

        when {
            params {
                // define parameters here. Example: 
                osdAccession = "OSD-205"
                gldsAccession = "GLDS-205"
                biomart_attribute = "AFFY ATH1 121501"
                limit_biomart_query = 300
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                            path("${ params.outputDir }/Metadata").list(),
                            path("${ params.outputDir }/00-RawData").list(),
                            path("${ params.outputDir }/01-oligo_NormExp").list(),
                            path("${ params.outputDir }/02-limma_DGE").list(),
                            ).match()
        }

    }

}