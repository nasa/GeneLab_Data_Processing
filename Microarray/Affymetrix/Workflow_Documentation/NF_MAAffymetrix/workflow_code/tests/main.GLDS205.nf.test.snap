{
    "GLDS-205: Should run without failures": {
        "content": [
            [
                "GLDS-205_metadata_GSE95388-ISA.zip:md5,d6b5dd590bcf3d6930581a419124c9e3",
                "OSD-205_microarray_v0_runsheet.csv:md5,c453cdbdfb93a2a99567b409477db302"
            ],
            [
                "raw_intensities_probe.csv:md5,9aef93a330160b4c7f6d87ea2315ccd7"
            ],
            [
                "normalized_expression_probeset.csv:md5,bf6ac705082742807dda479d69843507",
                "normalized_intensities_probe.csv:md5,e3130eaabe9cba2b0c99e442d955029c"
            ],
            [
                "SampleTable.csv:md5,16447f56ccdf5722640496640b53608e",
                "contrasts.csv:md5,d5e21359c412e7c04ed73e9ac1ef2571",
                "differential_expression.csv:md5,9285ff90870f67ee7cb33ce41235c39b",
                "visualization_PCA_table.csv:md5,811829d2a5521a831b8df0e97a6d2fb8"
            ]
        ],
        "timestamp": "2023-06-09T16:24:25+0000"
    }
}